pkg/security/ptracer/cws.go

```diff
@@ -108,9 +155,142 @@ func handleOpen(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg, regs
 		Mode:     uint32(tracer.ReadArgUint64(regs, 2)),
 	}
 
+	err = fillFileMetadata(filename, msg.Open)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func handleCreat(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs) error {
+	filename, err := tracer.ReadArgString(process.Pid, regs, 0)
+	if err != nil {
+		return err
+	}
+
+	filename, err = getFullPathFromFilename(process, filename)
+	if err != nil {
+		return err
+	}
+
+	msg.Type = ebpfless.SyscallTypeOpen
+	msg.Open = &ebpfless.OpenSyscallMsg{
+		Filename: filename,
+		Flags:    unix.O_CREAT | unix.O_WRONLY | unix.O_TRUNC,
+		Mode:     uint32(tracer.ReadArgUint64(regs, 1)),
+	}
+
+	err = fillFileMetadata(filename, msg.Open)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func handleNameToHandleAt(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs) error {
+	fd := tracer.ReadArgInt32(regs, 0)
+
+	filename, err := tracer.ReadArgString(process.Pid, regs, 1)
+	if err != nil {
+		return err
+	}
+
+	filename, err = getFullPathFromFd(process, filename, fd)
+	if err != nil {
+		return err
+	}
+
+	msg.Type = ebpfless.SyscallTypeOpen
+	msg.Open = &ebpfless.OpenSyscallMsg{
+		Filename: filename,
+	}
+	return nil
+}
+
+func handleNameToHandleAtRet(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs) {
+	if msg.Open == nil {
+		return
+	}
+
+	if ret := tracer.ReadRet(regs); ret < 0 {
+		return
+	}
+
+	pFileHandleData, err := tracer.ReadArgData(process.Pid, regs, 2, 8 /*sizeof uint32 + sizeof int32*/)
+	if err != nil {
+		return
+	}
+	var handleBytes uint32
+	var handleType int32
+	buf := bytes.NewReader(pFileHandleData[:4])
+	err = binary.Read(buf, native.Endian, &handleBytes)
+	if err != nil {
+		return
+	}
+	buf = bytes.NewReader(pFileHandleData[4:8])
+	err = binary.Read(buf, native.Endian, &handleType)
+	if err != nil {
+		return
+	}
+
+	key := fileHandleKey{
+		handleBytes: handleBytes,
+		handleType:  handleType,
+	}
+	fileHandleCache[key] = &fileHandleVal{
```

Comment:
should it be part of the Process cache and shared by threads like for file descriptors ?

Reply:
Suggestion applied